dbms/src/Common/TiFlashSecurity.h

```diff
@@ -167,6 +199,19 @@ struct TiFlashSecurityConfig
         }
         return result;
     }
+
+private:
+    String ca_path;
+    String cert_path;
+    String key_path;
+    LoggerPtr log;
+
+    bool redact_info_log = false;
+
+    std::set<String> allowed_common_names;
+
+    bool has_tls_config = false;
+    mutable std::mutex mu;
```

Comment:
Is this mutex necessary? Seems no member write except in the `init` method which is only called in ctor.

Reply:
> Is this mutex necessary? Seems no member write except in the `init` method which is only called in ctor.

Yes, when reloading config `init` will be called too.

Reply:
Saw it. Thanks~

Reply:
Maybe we can use a rw lock if we only re-init `TiFlashSecurityConfig` if it actually changes.